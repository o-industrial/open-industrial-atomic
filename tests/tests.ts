import './atoms/Action.test.tsx';
import './molecules/Modal.test.tsx';
import './templates/InspectorPanelTemplate.test.tsx';
import './utils/.tests.ts';
