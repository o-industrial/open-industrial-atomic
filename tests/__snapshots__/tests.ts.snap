export const snapshot = {};

snapshot[`Action > renders a primary solid action 1`] = `'<button type="button" class="-:inline-flex -:items-center -:justify-center -:font-semibold -:transition-all -:duration-200 -:ease-out -:px-4 -:py-2 -:bg-neon-violet-500 -:hover:bg-neon-violet-600 -:text-white -:border-neon-violet-500">Click to deploy</button>'`;

snapshot[`Action > renders an icon-only action 1`] = `'<button title="Close" aria-label="Close" type="button" class="-:inline-flex -:items-center -:justify-center -:font-semibold -:transition-all -:duration-200 -:ease-out -:rounded-full -:p-2 -:text-neon-indigo-400 -:hover:text-neon-indigo-300 -:border-neon-indigo-400"><svg aria-hidden="true" class="h-4 w-4"><use href="/icons/iconset#close" data-eac-bypass-base="true"></use></svg></button>'`;

snapshot[`Modal > renders a basic modal shell 1`] = `'<div role="presentation" class="fixed inset-0 z-50 bg-black/80 backdrop-blur-md flex items-center justify-center"><div role="dialog" aria-modal="true" aria-labelledby="modal-title-__id__" aria-describedby="modal-body-__id__" tabindex="-1" class="relative bg-neutral-900 border border-neutral-700 rounded-md shadow-xl overflow-hidden flex flex-col transition-all w-full max-w-5xl max-h-[90vh] m-4"><div class="flex items-center justify-between px-4 py-3 border-b border-neutral-700"><h2 id="modal-title-__id__" class="text-sm font-bold text-white uppercase tracking-wide w-full">Snapshot Modal</h2><button type="button" title="Close" aria-label="Close dialog" class="-:inline-flex -:items-center -:justify-center -:font-semibold -:transition-all -:duration-200 -:ease-out -:rounded-full -:p-2 -:text-neon-red-400 -:hover:text-neon-red-300 -:border-neon-red-400"><svg aria-hidden="true" class="h-4 w-4"><use href="/icons/iconset#close" data-eac-bypass-base="true"></use></svg></button></div><div id="modal-body-__id__" class="flex-1 overflow-y-auto p-4"><p>Declarative modal content</p></div></div></div>'`;

snapshot[`InspectorPanelTemplate > renders with close affordance 1`] = `'<aside role="complementary" aria-label="Inspector panel" aria-labelledby="inspector-heading-__id__" tabindex="-1" class="relative flex h-full w-full flex-col bg-neutral-900"><header class="sticky top-0 z-10 flex w-full items-center justify-between gap-3 border-b border-neutral-700 bg-neutral-800 px-4 py-2"><h2 id="inspector-heading-__id__" class="truncate text-sm font-semibold uppercase tracking-wide text-white">Structure Inspector</h2><button type="button" aria-label="Close inspector panel" title="Close inspector panel" class="-:inline-flex -:items-center -:justify-center -:font-semibold -:transition-all -:duration-200 -:ease-out -:rounded-full -:p-2 -:text-neon-indigo-400 -:hover:text-neon-indigo-300 -:border-neon-indigo-400"><svg aria-hidden="true" class="h-4 w-4"><use href="/icons/iconset#close" data-eac-bypass-base="true"></use></svg></button></header><div class="flex flex-1 flex-col gap-4 overflow-y-auto px-4 py-3"><div>Panel body content</div></div></aside>'`;
